test/unit/hint.test.js

```diff
@@ -0,0 +1,202 @@
+        { hints: [ new IndexHint('idx_id'), new IndexHint('idx_title'), new IndexHint('idx_title') ] }).toString(),
+        "UPDATE `articles` USE INDEX (idx_id,idx_title) SET `title` = 'hello', `gmt_modified` = '2017-12-12 00:00:00.000' WHERE `title` LIKE '%Post%' AND `gmt_deleted` IS NULL"
+      );
+
+      assert.equal(
+        Post.update({
+          title: { $like: '%Post%' }
+        },
+        { title: 'hello', updatedAt: date },
+        { hints: [ new IndexHint('idx_id'), { value: 'idx_title', type: HINT_TYPE.INDEX_HINT }, new IndexHint('idx_title') ] }).toString(),
+        "UPDATE `articles` USE INDEX (idx_id,idx_title) SET `title` = 'hello', `gmt_modified` = '2017-12-12 00:00:00.000' WHERE `title` LIKE '%Post%' AND `gmt_deleted` IS NULL"
+      );
+
+      assert.equal(
+        Post.update({
+          title: { $like: '%Post%' }
+        },
+        { title: 'hello', updatedAt: date },
+        { hints: [ new IndexHint('idx_id'), { value: 'idx_title', type: INDEX_HINT_TYPE.USE }, new IndexHint('idx_title') ] }).toString(),
+        "UPDATE `articles` USE INDEX (idx_id,idx_title) SET `title` = 'hello', `gmt_modified` = '2017-12-12 00:00:00.000' WHERE `title` LIKE '%Post%' AND `gmt_deleted` IS NULL"
+      );
+
+      assert.equal(
+        Post.update({
+          title: { $like: '%Post%' } },
+          { title: 'hello', updatedAt: date },
+          {
+            hints: [
+              new IndexHint('idx_id', INDEX_HINT_TYPE.FORCE),
+              new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
+              new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
+              new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
+              new IndexHint('idx_hle1', INDEX_HINT_TYPE.IGNORE),
+              new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
+              new IndexHint('idx_hle2', INDEX_HINT_TYPE.USE),
+            ]
+          }).toString(),
+          "UPDATE `articles` FORCE INDEX (idx_id,idx_title) IGNORE INDEX (idx_hle,idx_hle1) USE INDEX (idx_hle2) SET `title` = 'hello', `gmt_modified` = '2017-12-12 00:00:00.000' WHERE `title` LIKE '%Post%' AND `gmt_deleted` IS NULL"
+      );
+    });
+  });
+
+  it('mixing hint', () => {
+    const date = new Date(2017, 11, 12);
+
+    assert.equal(
+      Post.update({
+        title: { $like: '%Post%' } },
+        { title: 'hello', updatedAt: date },
+        {
+          hints: [
+            new IndexHint('idx_id', INDEX_HINT_TYPE.FORCE),
+            new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
+            new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
+            new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
+            new IndexHint('idx_hle1', INDEX_HINT_TYPE.IGNORE),
+            new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
+            new IndexHint('idx_hle2', INDEX_HINT_TYPE.USE),
+            new Hint('SET_VAR(foreign_key_checks=OFF)'),
+            new Hint('MAX_EXECUTION_TIME(1000)')
+          ]
+        }).toString(),
```

Comment:
The indentation of this block looks a bit weird. Suggestion:

```js
Post.update(
  { title: { $like: '%Post%' } },
  { title: 'hello', updatedAt: date },
  {
    hints: [
      new IndexHint('idx_id', INDEX_HINT_TYPE.FORCE),
      new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
      new IndexHint('idx_title', INDEX_HINT_TYPE.FORCE),
      new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
      new IndexHint('idx_hle1', INDEX_HINT_TYPE.IGNORE),
      new IndexHint('idx_hle', INDEX_HINT_TYPE.IGNORE),
      new IndexHint('idx_hle2', INDEX_HINT_TYPE.USE),
      new Hint('SET_VAR(foreign_key_checks=OFF)'),
      new Hint('MAX_EXECUTION_TIME(1000)')
    ]
  }
).toString(),
```

Reply:
Personally I'd prefer method chaining style like Active Record https://bigbinary.com/blog/rails-6-supports-optimizer-hints

```js
Post.update({ title: { $like: '%Post%' } }, { title: 'hello', updatedAt: date })
  .optimizerHints('SET_VAR(foreign_key_checks=OFF)', 'MAX_EXECUTION_TIME(1000)')
  .useIndex('idx_hle2')
  .forceIndex('idx_id', 'idx_title', 'idx_title')
  .ignoreIndex('idx_hle', 'idx_hle1', 'idx_hle')
```